ASoC: SOF: loader: print DSP boot time in debug prints

kv2019i opened the pull request.

ujfalusi commented on sound/soc/sof/loader.c:
```diff
@@ -185,9 +188,12 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 
-	return 0;
+	dev_dbg(sdev->dev, "firmware boot (post) complete, took %lldus\n",
+		ktime_us_delta(ktime_get(), boot_start));
```
I think this is pointless to print out if the ops->post_fw_boot is not set, I would move it inside of the if() and print it only if `ret == 0` with "Full firmware startup took %lldus\n".
Or you might want to break it down to firmware boot proper and post boot time.

kv2019i replied:
I started making these changes, but then changed my mind.  post_fw_boot has the additional steps on Intel (IPC4) platforms, but there are also extra steps in snd_sof_dsp_post_fw_run(), and given these prints are added to common code, shouldn't we cover all the post boot steps in the measured time (DSP post_boot and IPC-version dependent post_boot)...?

Or is this too Intel specific and we should hide these prints to some more HW specific code?  @dbaluta , what do you think?

It would still seem useful information to have (all platforms need to boot the FW).

ujfalusi replied:
I still think that this is pointless to print if `sdev->ipc->ops->post_fw_boot == NULL`

Also, a theoretical log:
```
firmware boot complete, took 100us
firmware boot (post) complete, took 140us
```
the full boot time is 100us+140us or 140us? I know the answer, but from the print I assume the former, while the later is the meaning.

and w/o post_fw_boot:
```
firmware boot complete, took 100us
firmware boot (post) complete, took 101us
```
The printk took 1us, is that really something which helps?

ujfalusi commented on sound/soc/sof/loader.c:
```diff
@@ -113,8 +113,8 @@ EXPORT_SYMBOL(snd_sof_load_firmware_memcpy);
-                .state = SOF_DSP_PM_D0,
-        };
+		.state = SOF_DSP_PM_D0,
+	};
```
this should be the first patch.

kv2019i replied:
Uh, sorry, I put this last on purpose. Now fixed in updated series.

kv2019i committed:
ASoC: SOF: loader: fix indentation in snd_sof_run_firmware()

Fix to use tabs for indentation.

Signed-off-by: Kai Vehmanen <kai.vehmanen@linux.intel.com>

kv2019i committed:
ASoC: SOF: loader: print DSP boot time in debug prints

DSP boot time is a key contributing factor to audio device set up time
with SOF. Add measurement of boot time and include it in the debug
messages.

Move the "firmware boot complete" message to end of the run_firmware()
function and include possible post_fw_boot() actions in the boot up
time.

Signed-off-by: Kai Vehmanen <kai.vehmanen@linux.intel.com>

ujfalusi commented:
> @ujfalusi @lyakh Is this useful addition or shall we close this?

I think this is useful, but needs some clarification.

kv2019i commented:
New revision:
- simplified the patch as per feedback from @ujfalusi

ujfalusi commented on sound/soc/sof/loader.c:
```diff
@@ -185,9 +186,13 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
-	return 0;
+	if (!ret)
+		dev_dbg(sdev->dev, "firmware boot complete, took %lldus\n",
+			ktime_us_delta(ktime_get(), boot_start));
```
Are you fine with logging now the combined snd_sof_dsp_run() + snd_sof_dsp_post_fw_run() + sdev->ipc->ops->post_fw_boot() ?

Note: the basefw boot time can be deduced based on the timestamps in kernel log:
```
"booting DSP firmware"
...
"fw_state change: 6 -> 7"
```

ujfalusi commented on sound/soc/sof/loader.c:
```diff
@@ -171,7 +173,6 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 	if (sdev->fw_state == SOF_FW_BOOT_READY_FAILED)
 		return -EIO; /* FW boots but fw_ready op failed */
 
-	dev_dbg(sdev->dev, "firmware boot complete\n");
```
The firmware boot proper is done at this point to be pragmatic, from this point we have the post boot calls, which needs a running firmware on the DSP side...
